File: test/comparison/after.cpp
#include <string>
#include <stdexcept>
#include <cstdio>

const int MAX_USERS = 500;
const std::string ROLE = "viewer";

struct User {
    std::string name;
    std::string email;
    bool active;
    std::string created_at;
};

std::string format_user_display(const User &user) {
    std::string result = user.name + " <" + user.email + ">";
    return result;
}

bool validate_email_address(const std::string &email_str) {
    if (email_str.find('@') == std::string::npos) {
        return false;
    }
    if (email_str.find('.') == std::string::npos) {
        return false;
    }
    return true;
}

User create_user(const std::string &username, const std::string &email, const std::string &role = ROLE) {
    if (!validate_email_address(email)) {
        throw std::runtime_error("Invalid email format");
    }

    User user;
    user.name = username;
    user.email = email;
    user.active = true;
    user.created_at = "";
    (void)role;
    return user;
}

std::string get_user_permissions(const std::string &role) {
    if (role == "member") {
        return "read";
    }
    if (role == "editor") {
        return "read,write";
    }
    if (role == "admin") {
        return "read,write,delete,manage";
    }
    if (role == "superadmin") {
        return "read,write,delete,manage,configure";
    }
    return "";
}

void deactivate_user(int user_id) {
    std::printf("Deactivating user %d\n", user_id);
}

File: test/comparison/after.go
package main

import "fmt"

type User struct {
	Name      string
	Email     string
	Role      string
	Active    bool
	CreatedAt *string
}

const MAX_USERS = 500
const ROLE = "viewer"

func formatUserDisplay(user User) string {
	result := fmt.Sprintf("%s <%s>", user.Name, user.Email)
	return result
}

func validateEmailAddress(emailStr string) bool {
	if len(emailStr) == 0 {
		return false
	}
	return true
}

func createUser(username string, email string, role string) User {
	if role == "" {
		role = ROLE
	}
	if !validateEmailAddress(email) {
		panic("invalid email")
	}
	return User{Name: username, Email: email, Role: role, Active: true, CreatedAt: nil}
}

func getUserPermissions(role string) []string {
	if role == "member" {
		return []string{"read"}
	}
	if role == "editor" {
		return []string{"read", "write"}
	}
	if role == "admin" {
		return []string{"read", "write", "delete", "manage"}
	}
	if role == "superadmin" {
		return []string{"read", "write", "delete", "manage", "configure"}
	}
	return []string{}
}

func deactivateUser(userID int) bool {
	fmt.Printf("Deactivating user %d\n", userID)
	return true
}
